pyartcd/pyartcd/pipelines/quay_doomsday_backup.py

```diff
@@ -44,35 +48,66 @@ def sync_arch(self, arch: str):
+        # because cmd_assert_async does not have retry logic
+        retry = AsyncRetrying(reraise=True, stop=stop_after_attempt(N_RETRIES))
+        try:
+            self.runtime.logger.info("[%s] Running mirror command: %s", arch, mirror_cmd)
+            await retry(cmd_assert_async, mirror_cmd)
+            self.runtime.logger.info("[%s] Mirror command ran successfully", arch)
+            if self.runtime.dry_run:
+                self.runtime.logger.info("[DRY RUN] [%s] Would have run %s", arch, " ".join(aws_cmd))
+                self.runtime.logger.info("[DRY RUN] [%s] Would have messaged slack", arch)
+            else:
+                sleep(5)
+                self.runtime.logger.info("[%s] Running aws command: %s", arch, aws_cmd)
+                await retry(cmd_assert_async, aws_cmd)
+                self.runtime.logger.info("[%s] AWS command ran successfully", arch)
+                sleep(5)
+
+                await self.slack_client.say_in_thread(f":white_check_mark: Successfully synced {self.version}-{arch}")
+
+        except ChildProcessError as e:
+            self.runtime.logger.error("[%s] Failed to sync: %s", arch, e)
+            if self.runtime.dry_run:
+                self.runtime.logger.info("[DRY RUN] [%s] Would have messaged slack", arch)
+            else:
+                await self.slack_client.say_in_thread(f":warning: Failed to sync {self.version}-{arch}: {e}")
+            return False
+
+        if os.path.exists(f"{self.workdir}/{path}"):
+            self.runtime.logger.info("[%s] Cleaning dir: %s", arch, f"{self.workdir}/{path}")
+            shutil.rmtree(f"{self.workdir}/{path}")
+
+        return True
+
+    async def run(self) -> None:
+        mkdirs(self.workdir)
+
+        if not self.runtime.dry_run:
+            await self.slack_client.say_in_thread(f":construction: Syncing arches {', '.join(self.arches)} of {self.version} to AWS S3 Bucket :construction:")
         else:
-            sleep(5)
-            self.runtime.logger.info("Running aws command: %s", aws_cmd)
-            cmd_assert(aws_cmd, retries=N_RETRIES)
-            self.runtime.logger.info("AWS command ran successfully")
-            sleep(5)
+            self.runtime.logger.info("[DRY RUN] Would have messaged slack")
 
-        if os.path.exists(path):
-            self.runtime.logger.info("Cleaning dir: %s", path)
-            shutil.rmtree(path)
+        tasks = [self.sync_arch(arch) for arch in self.arches]
+        results = await asyncio.gather(*tasks)
 
-    def run(self):
-        mkdirs(self.workdir)
-
-        for arch in self.arches:
-            self.runtime.logger.info("Now syncing arch %s", arch)
-            self.sync_arch(arch)
+        # Report the results to Slack
+        if not self.runtime.dry_run:
+            if all(results):
+                await self.slack_client.say_in_thread(":done_it_is: All arches synced successfully", broadcast=True)
```

Comment:
You are right, thank you. 
I've added a new method `add_reaction` to `slack.py` which is then used to add :done_it_is: on the main message (I find :done_it_is: to be clearer than :done:) like [this](https://redhat-internal.slack.com/archives/C05HB1F1THP/p1733752159971629)
However, the underlying Slack api function [reactions_add](https://api.slack.com/methods/reactions.add) requires channel ID instead of just channel name and getting the ID is a little bit tricky.
I've added another method (`get_channel_id`) to `slack.py` which tries to get the ID of the bound channel. It only works for private channels for now because there's too many public channels and we hit request ratelimit while searching through them.